mysql-test/suite/innodb/t/ts_encrypt.test

```diff
@@ -0,0 +1,174 @@
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE pt1 ADD PARTITION (PARTITION p5 VALUES LESS THAN (10) TABLESPACE ts_encrypted);
+ALTER TABLE pt1 ADD PARTITION (PARTITION p6 VALUES LESS THAN (12) TABLESPACE innodb_system);
+
+ALTER TABLE pt2 ADD PARTITION (PARTITION p4 VALUES LESS THAN (8) TABLESPACE ts_encrypted_new);
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE pt2 ADD PARTITION (PARTITION p5 VALUES LESS THAN (10) TABLESPACE ts_unencrypted);
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE pt2 ADD PARTITION (PARTITION p6 VALUES LESS THAN (12) TABLESPACE innodb_system);
+
+# see bug https://bugs.mysql.com/bug.php?id=86809
+
+# --error ER_ILLEGAL_HA_CREATE_OPTION
+# CREATE TABLE pt3 (a INT NOT NULL, PRIMARY KEY(a))
+# 	ENGINE=InnoDB TABLESPACE ts_encrypted ENCRYPTION='y'
+# 	PARTITION BY RANGE (a) PARTITIONS 3 (
+# 		PARTITION p1 VALUES LESS THAN (2),
+# 		PARTITION p2 VALUES LESS THAN (4) TABLESPACE innodb_file_per_table,
+# 		PARTITION p3 VALUES LESS THAN (6) TABLESPACE innodb_system);
+
+# --error ER_ILLEGAL_HA_CREATE_OPTION
+# CREATE TABLE pt4 (a INT NOT NULL, PRIMARY KEY(a))
+# 	ENGINE=InnoDB TABLESPACE ts_encrypted ENCRYPTION='y'
+# 	PARTITION BY RANGE (a) PARTITIONS 3 (
+# 		PARTITION p1 VALUES LESS THAN (2),
+# 		PARTITION p2 VALUES LESS THAN (4) TABLESPACE innodb_file_per_table,
+# 		PARTITION p3 VALUES LESS THAN (6) TABLESPACE ts_unencrypted);
+
+# --error ER_ILLEGAL_HA_CREATE_OPTION
+# CREATE TABLE pt5 (a INT NOT NULL, PRIMARY KEY(a))
+# 	ENGINE=InnoDB TABLESPACE ts_unencrypted ENCRYPTION='y'
+# 	PARTITION BY RANGE (a) PARTITIONS 3 (
+# 		PARTITION p1 VALUES LESS THAN (2),
+# 		PARTITION p2 VALUES LESS THAN (4) TABLESPACE innodb_file_per_table,
+# 		PARTITION p3 VALUES LESS THAN (6) TABLESPACE ts_encrypted);
+
+# SELECT FILE_ID, FILE_NAME FROM INFORMATION_SCHEMA.FILES
+# WHERE FILE_TYPE = 'TABLESPACE' ORDER BY FILE_ID;
+
+# SELECT NAME, FLAG, SPACE_TYPE FROM INFORMATION_SCHEMA.INNODB_SYS_TABLESPACES;
+
+
+# Make sure that innodb_strict_mode doesn't matter
+SET GLOBAL innodb_file_per_table=OFF;
+SET GLOBAL innodb_strict_mode=OFF;
+
+# innodb_strict_mode change to take effect
+disconnect default;
+connect(default,localhost,root,,test);
+
+CREATE TABLE q1 (c1 int);
+
+--error ER_ILLEGAL_HA_CREATE_OPTION
+ALTER TABLE q1 ENCRYPTION='Y',ALGORITHM=COPY;
+
+eval SET GLOBAL innodb_file_per_table=$innodb_file_per_table;
+eval SET GLOBAL innodb_strict_mode=$innodb_strict_mode;
+
+# innodb_strict_mode change to take effect
+disconnect default;
+connect(default,localhost,root,,test);
+
+# cleanup
+DROP TABLE t2;
+DROP TABLE t3;
+DROP TABLE t6;
+DROP TABLE t7;
+DROP TABLE t9;
+DROP TABLE t10;
+DROP TABLE t11;
+DROP TABLE t12;
+DROP TABLE q1;
+DROP TABLE p1;
+DROP TABLE p2;
+
+DROP TABLE pt1;
+DROP TABLE pt2;
+
+SELECT TABLE_NAME, PARTITION_NAME, SUBPARTITION_NAME, TABLESPACE_NAME
+FROM INFORMATION_SCHEMA.PARTITIONS WHERE TABLESPACE_NAME='ts_encrypted';
```

Comment:
Although the result set should be empty, add `ORDER BY` here as well (just in case).